vnext/Mso/src/dispatchQueue/looperScheduler.cpp

```diff
@@ -44,35 +44,37 @@ LooperScheduler::~LooperScheduler() noexcept {
   for (;;) {
     if (auto self = weakSelf.GetStrongPtr()) {
       if (auto queue = self->m_queue.GetStrongPtr()) {
-        DispatchQueue q{Mso::CntPtr(queue)};
+        auto &taskStarting = self->m_settings.TaskStarting;
+        auto &taskCompleted = self->m_settings.TaskCompleted;
         DispatchTask task;
         while (queue->TryDequeTask(task)) {
-          if (auto &func = self->m_settings.TaskStarting) {
-            func(q);
+          if (taskStarting) {
+            taskStarting(DispatchQueue{Mso::CntPtr(queue)});
```

Comment:
>Mso::CntPtr [](http://example.com/codeflow?start=39&length=11)

Each call to Mso::CntPtr like this causes a new interlocked increment/decrement. Please see the code that I suggested before to reduce number of interlocked increments/decrements. The code was showing how to do it only once before the `while` loop.
We do not want to do it for each iteration.
If you would like to optimize it, then it can be done completely on-demand:
```
DispatchQueue q{nullptr};
while (queue->TryDequeTask(task)) {
   if (auto& func = self->m_settings.TaskStarting) {
    q = q ? std::move(q) : DispatchQueue{Mso::CntPtr(q)};
    func(q);
  }

  queue->InvokeTask(std::move(task), std::nullopt);

  if (auto& func = self->m_settings.TaskCompleted) {
    q = q ? std::move(q) : DispatchQueue{Mso::CntPtr(q)};
    func(q);
  }
}

Reply:
Done!